### PR TITLE
Schemas data loading

### PR DESCRIPTION
**Pull request by grechaw**
This task adds support for loading schemas database to ml-gradle.
Let's see if this new batch of PRs works for you -- I've been having success so far.

### CODE REVIEW

**Comment by rjrudin** [comment]
@grechaw Sorry for the delay, been swamped, I'll get this merged in and publish 2.1 by the end of this coming weekend.

File: src/main/groovy/com/marklogic/gradle/MarkLogicPlugin.groovy [hunk 1]
```diff
@@ -181,6 +183,10 @@ class MarkLogicPlugin implements Plugin<Project> {
         project.task("mlWatch", type: WatchTask, group: modulesGroup, description: "Run a loop that checks for new/modified modules every second and loads any that it finds")
         project.task("mlDeleteModuleTimestampsFile", type: DeleteModuleTimestampsFileTask, group: modulesGroup, description: "Delete the properties file in the build directory that keeps track of when each module was last loaded")
 
+        String schemasGroup = "ml-gradle Schemas"
+        project.task("mlLoadSchemas", type: LoadSchemasTask, group: schemasGroup, description: "Loads special-purpose data into the schemas database (XSD schemas, Inference rules, and [MarkLogic 9] Extraction Templates)")
+        project.task("mlReloadSchemas", type: LoadSchemasTask, dependsOn: ["mlClearSchemasDatabase", "mlLoadSchemas"], group: schemasGroup, description: "Clears schemas database then loads special-purpose data into the schemas database (XSD schemas, Inference rules, and [MarkLogic 9] Extraction Templates)")
```
Comment (rjrudin) [inline src/main/groovy/com/marklogic/gradle/MarkLogicPlugin.groovy]: For mlReloadSchemas, you actually don't want a type - look at mlReloadModules as an example. 

Also, the order of tasks in dependsOn doesn't force when Gradle will invoke the tasks. To ensure that mlLoadSchemas always runs after mlClearSchemasDatabase, you do this:

project.task("mlLoadSchemas", type....).mustRunAfter("mlClearSchemasDatabase")

I think mlLoadSchemas has to be defined after mlSchemasDatabase in order for mustRunAfter to work properly.

File: src/main/groovy/com/marklogic/gradle/task/schemas/LoadSchemasTask.groovy [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+class LoadSchemasTask extends MarkLogicTask {
+
+    @TaskAction
+    void loadSchemas() {
+       LoadSchemasCommand command = null
+        // Check for a LoadSchemasCommand in the AppDeployer first, as that may have additional configuration that we
+        // don't want to have to duplicate on this task
+        AppDeployer d = getAppDeployer()
+        if (d instanceof SimpleAppDeployer) {
+            command = d.getCommand("mlLoadSchemasCommands")
```
Comment (rjrudin) [inline src/main/groovy/com/marklogic/gradle/task/schemas/LoadSchemasTask.groovy]: getCommand is a shortcut that allows you to fetch a command, from the list of commands in SimpleAppDeployer, based on its class name (not including its package). So the input here should be "LoadSchemasCommand".

**Comment by grechaw** [comment]
@rjrudin I'm not confident that the default path property is working for ml-schemas.  When you give it a test run please note if I've done something wrong there.  It can be a bit of a time suck trying to get all these bits right.

**Comment by rjrudin** [comment]
I merged it into dev and will test it out with some other changes for 2.1.0

**Pull request closed without merging**